Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -214,10 +215,16 @@ extension BrowserViewController: WKNavigationDelegate {
                 off.compactMap { $0.rule }.forEach(controller.remove)
               
                 if let tab = tabManager[webView] {
-                    tab.userScriptManager?.isFingerprintingProtectionEnabled = domainForShields.isShieldExpected(.FpProtection)
+                    tab.userScriptManager?.isFingerprintingProtectionEnabled =
+                        !domainForShields.isAllShieldsOff()
+                        &&
+                        domainForShields.isShieldExpected(.FpProtection)
```

Comment:
Wouldn't it be better to put `!domainForShields.isAllShieldsOff()` logic inside of `isShieldExpected`?

It makes more sense and we can still exclude it for safe browsing.

Now it's easy to forget to add `isAllShieldsOff` for all cases

Reply:
I tried that, it breaks the UI for some reason. The toggles actually show the wrong state on initial opening of the panel if I do that because this function isn't just for the shields, it's used for the UI as well.